core/src/sumeragi/main_loop.rs

```diff
@@ -605,16 +593,34 @@ fn handle_message<F: FaultInjection>(
         (Message::ViewChangeSuggested, _) => {
             trace!("Received view change suggestion.");
         }
-        (Message::BlockSyncUpdate(BlockSyncUpdate { block }), _) => {
+        (Message::BlockSyncUpdate(block_sync_update), _) => {
+            let block = block_sync_update.extract_block_unchecked();
             info!(%addr, %role, hash=%block.hash(), "Block sync update received");
 
-            let block = match block.revalidate(current_topology) {
-                Ok(block) => block,
-                Err((block, error)) => {
-                    error!(%addr, %role, hash=%block.hash(), ?error, "Block not valid");
+            let block_hash = block.hash();
+            if block
+                .revalidate::<true>(
```

Comment:
This is not genesis transaction so must be

```suggestion
                .revalidate::<false>(
```